pkg/config/config.go

```diff
@@ -205,6 +205,16 @@ func (c *ScrapeConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		unmarshalled.ProfilingConfig = defaults.ProfilingConfig
 	} else if unmarshalled.ProfilingConfig.PprofConfig == nil {
 		unmarshalled.ProfilingConfig.PprofConfig = defaults.ProfilingConfig.PprofConfig
+	} else {
+		// Merge unmarshalled config with defaults
+		for pt := range unmarshalled.ProfilingConfig.PprofConfig {
```

Comment:
Before this section was removed the code was like this. 

```suggestion
		for pt := range defaults.ProfilingConfig.PprofConfig {
			// nothing set yet so simply use the default
			if unmarshalled.ProfilingConfig.PprofConfig[pt] == nil {
				unmarshalled.ProfilingConfig.PprofConfig[pt] = pc
				continue
			}
```

Maybe we want that again, but that means that we always add all defaults and then rather disable the ones we don't want?! I suppose that was the original behavior, though.